src/fideslang/models.py

```diff
@@ -358,14 +500,16 @@ class DatasetMetadata(BaseModel):
     """
     The DatasetMetadata resource model.
 
-    Object used to hold application specific metadata for a dataset
+    Object used to hold application specific metadata for a dataset,
+    including annotations to help with query traversal.
```

Comment:
```suggestion
    including annotations to help with query traversal.
```

Nit: I'm not sure we need to specify the usage of query traversal here, as in the future we could use this field for any number of things, so specifying a single current use-case seems a bit odd